src/asset/MountArchive.cpp

```diff
@@ -0,0 +1,89 @@
+#include "MountArchive.h"
+
+#include "bit7z/bitarchivereader.hpp"
+#include "utils/Log.h"
+
+#include <cstddef>
+#include <filesystem>
+#include <format>
+#include <string>
+#include <vector>
+
+MountArchive::MountArchive(std::filesystem::path archive_path)
+    : Mount(archive_path), reader(new bit7z::BitArchiveReader(library, path.string())) {
+}
+
+MountArchive::~MountArchive() {
+    MountArchive::save_cache();
+}
+
+Mount::MountType MountArchive::get_type() {
+    return ARCHIVE;
+}
+
+bool MountArchive::load() {
+    reset_reader();
+
+    try {
+        reader = new bit7z::BitArchiveReader(library, get_path().string());
+        reader->test();
+    }
+    catch (const std::exception& exception) {
+        return false;

```

Comment:
do we need this broad of an exception catch? should we handle the error more gracefully? why not propagate the exception up the stack? should probably rethrow here.